### PR TITLE
Fork suicide

### PR DESCRIPTION
**Pull request by Varelion**
## What Does This PR Do

Adds a suicide option for the fork.

## Why It's Good For The Game
 Adds one more suicide option, in the form of the fork, to the myriad list of other suicides. This is also a reference to a well-known meme of the early '10s.

## Testing
 Clean compile and ran server.

## Changelog
:cl:
add: fork suicide
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/kitchen.dm [hunk 1]
```diff
@@ -70,6 +70,20 @@
...
+/obj/item/kitchen/utensil/fork/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is putting the [src] into the garbage disposal!</span>")
+	var/message = "DING DING DING DING DING DING DING DING DING"
+	var/span = "reallybig"
+	sleep(2 SECONDS)
+	user.emote("scream")
+	playsound(src.loc, 'sound/machines/juicer.ogg',50, TRUE, -1)
+	animate_fading_leap_up(user)
+	user.create_chat_message(user, message, FALSE, "big")
+	audible_message("<span class='[span]'>\"[message]\"</span>", hearing_distance = 14)
```
Comment (DGamerL) [inline code/game/objects/items/weapons/kitchen.dm:82]: Use `visible_message` here instead

File: code/game/objects/items/weapons/kitchen.dm [hunk 2]
```diff
@@ -70,6 +70,20 @@
 	desc = "It's a fork. Sure is pointy."
 	icon_state = "fork"
 
+/obj/item/kitchen/utensil/fork/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is putting the [src] into the garbage disposal!</span>")
+	var/message = "DING DING DING DING DING DING DING DING DING"
+	var/span = "reallybig"
+	sleep(2 SECONDS)
```
Comment (GDNgit) [inline code/game/objects/items/weapons/kitchen.dm:77]:
```suggestion
	if(!do_mob(user, user, 2 SECONDS) || QDELETED(user))
		return SHAME
```

File: code/game/objects/items/weapons/kitchen.dm [hunk 3]
```diff
@@ -70,6 +70,20 @@
 	desc = "It's a fork. Sure is pointy."
 	icon_state = "fork"
 
+/obj/item/kitchen/utensil/fork/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is putting the [src] into the garbage disposal!</span>")
```
Comment (GDNgit) [inline code/game/objects/items/weapons/kitchen.dm:74]:
```suggestion
	user.visible_message("<span class='suicide'>[user] is putting the [src] into the garbage disposal! It looks like [user.p_theyre()] trying to commit suicide!</span>")
```
All modern suicide acts use this format

File: code/game/objects/items/weapons/kitchen.dm [hunk 4]
```diff
@@ -70,6 +70,20 @@
...
+	var/message = "DING DING DING DING DING DING DING DING DING"
+	var/span = "reallybig"
+	sleep(2 SECONDS)
+	user.emote("scream")
+	playsound(src.loc, 'sound/machines/juicer.ogg',50, TRUE, -1)
```
Comment (GDNgit) [inline code/game/objects/items/weapons/kitchen.dm:79]:
```suggestion
	playsound(get_turf(src), 'sound/machines/juicer.ogg',50, TRUE, -1)
```

File: code/game/objects/items/weapons/kitchen.dm [hunk 5]
```diff
@@ -70,6 +70,20 @@
...
+	animate_fading_leap_up(user)
+	user.create_chat_message(user, message, FALSE, "big")
+	audible_message("<span class='[span]'>\"[message]\"</span>", hearing_distance = 14)
+	sleep(1 SECONDS)
+	user.gib()
```
Comment (GDNgit) [inline code/game/objects/items/weapons/kitchen.dm:84]:
```suggestion
	addtimer(CALLBACK(src, PROC_REF(gib)), 1 SECONDS)
```

File: code/game/objects/items/weapons/kitchen.dm (hunk 2 above)
Comment (Varelion) [reply code/game/objects/items/weapons/kitchen.dm:77]: Not a fan of this one, it's making the code less readable.

File: code/game/objects/items/weapons/kitchen.dm [hunk 6]
```diff
@@ -70,6 +70,20 @@
 	desc = "It's a fork. Sure is pointy."
 	icon_state = "fork"
 
+/obj/item/kitchen/utensil/fork/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is putting the [src] into the garbage disposal! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	var/message = "DING DING DING DING DING DING DING DING DING"
+	var/span = "reallybig"
+	sleep(2 SECONDS)
+	user.emote("scream")
+	playsound(get_turf(src), 'sound/machines/juicer.ogg',50, TRUE, -1)
```
Comment (Burzah) [inline code/game/objects/items/weapons/kitchen.dm:79]:
```suggestion
	playsound(get_turf(src), 'sound/machines/juicer.ogg', 50, TRUE, -1)
```

File: code/game/objects/items/weapons/kitchen.dm (hunk 2 above)
Comment (Henri215) [reply code/game/objects/items/weapons/kitchen.dm:77]: What is the problem with it? Hes not rewriting the same thing in another way.

Comment (DGamerL) [reply code/game/objects/items/weapons/kitchen.dm:77]: As Henri said, this gives the progressbar and makes it cancellable, alongside a check if the mob is qdeleted. It does more than just waiting 2 seconds

File: code/game/objects/items/weapons/kitchen.dm [hunk 7]
```diff
@@ -70,6 +70,20 @@
 	desc = "It's a fork. Sure is pointy."
 	icon_state = "fork"
 
+/obj/item/kitchen/utensil/fork/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is putting the [src] into the garbage disposal! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	var/message = "DING DING DING DING DING DING DING DING DING"
+	var/span = "reallybig"
```
Comment (Sirryan2002) [inline code/game/objects/items/weapons/kitchen.dm:76]: i would just put this directly into the audible message string

**Review by S34NW — CHANGES_REQUESTED** [review]
Please don't just ignore reviews from team members and mark them as resolved. Implement the suggested changes above.

**Comment by Varelion** [comment]
> Please don't just ignore reviews from team members and mark them as resolved. Implement the suggested changes above.

Will do. On one of them though, I don't think there is precedent for suicide progress bars, or suicide interruption though.

**Comment by Denthamos** [comment]
Heh, we should also get one with the spoon where when you suicide with it, it removes your heart from your body too.  As a reference to Kevin Costners Robin Hood.

**Comment by Varelion** [comment]
> Heh, we should also get one with the spoon where when you suicide with it, it removes your heart from your body too. As a reference to Kevin Costners Robin Hood.

THAT'S GLORIOUS, I'll do it.

File: code/game/objects/items/weapons/kitchen.dm (hunk 7 above)
Comment (Varelion) [reply code/game/objects/items/weapons/kitchen.dm:76]: Done

File: code/game/objects/items/weapons/kitchen.dm (hunk 1 above)
Comment (Varelion) [reply code/game/objects/items/weapons/kitchen.dm:82]: done

File: code/game/objects/items/weapons/kitchen.dm (hunk 5 above)
Comment (Varelion) [reply code/game/objects/items/weapons/kitchen.dm]: Implemented.

**Comment by Varelion** [comment]
@S34NW Alright, how is this?

File: code/game/objects/items/weapons/kitchen.dm [hunk 8]
```diff
@@ -70,6 +70,18 @@
...
 	icon_state = "fork"
 
+/obj/item/kitchen/utensil/fork/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is putting the [src] into the garbage disposal! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	sleep(2 SECONDS)
+	user.emote("scream")
+	playsound(get_turf(src), 'sound/machines/juicer.ogg', 50, TRUE, -1)
+	animate_fading_leap_up(user)
+	user.create_chat_message(user, "DING DING DING DING DING DING DING DING DING", FALSE, "big")
+	visible_message("<span class='reallybig'>\"DING DING DING DING DING DING DING DING DING\"</span>", hearing_distance = 14)
```
Comment (lewcc) [inline code/game/objects/items/weapons/kitchen.dm:80]: This is failing CI, please fix this.

**Review by lewcc — CHANGES_REQUESTED** [review]

**Comment by S34NW** [comment]
Please see the above review within a week or this PR will be closed due to inactivity

**Comment by S34NW** [comment]
No changes in the requested timeframe, closing due to being stale.
Please feel free to make a new PR with this once the requested changes have been implemented, you can also reach out to our coding community for help at https://discord.gg/paradisess13

**Pull request closed without merging**